core/include/bitarithm.h

```diff
@@ -136,8 +136,33 @@ static inline uint8_t bitarithm_bits_set_u32(uint32_t v)
 uint8_t bitarithm_bits_set_u32(uint32_t v);
 #endif
 
+/**
+ * @brief   Returns the number of the highest '1' bit in a value
+ *
+ *          Internal software implementation for 32 bit platforms,
+ *          use @see bitarithm_msb in application code.
+ * @param[in]   v   Input value
+ * @return          Bit Number
+ */
+unsigned bitarith_msb_32bit_no_native_clz(unsigned v);
+
 /* implementations */
 
+static inline unsigned bitarithm_msb(unsigned v)
+{
+#if defined(BITARITHM_HAS_CLZ)
+    return 8 * sizeof(v) - __builtin_clz(v) - 1;
+#elif ARCH_32_BIT
+    return bitarith_msb_32bit_no_native_clz(v);
+#else
+    unsigned r = 0;
+    while (v >>= 1) { /* unroll for more speed... */
+        ++r;
+    }
+    return r;
+#endif
```

Comment:
```suggestion
    if (IS_ACTIVE(BITARITHM_HAS_CLZ)) {
        return 8 * sizeof(v) - __builtin_clz(v) - 1;
    }
    else if (IS_ACTIVE(ARCH_32_BIT)) {
        return bitarith_msb_32bit_no_native_clz(v);
    }
    unsigned r = 0;
    while (v >>= 1) { /* unroll for more speed... */
        ++r;
    }
    return r;
```

Reply:
Cool! I thought HTTP is build on top of TCP, so that communication is done reliable and duplicated packets get de-duplicated on the transport layer. Apparently I missed something. (Resolving this, it is enough to have the comment posted once.)